CoreDistributedCache/NHibernate.Caches.CoreDistributedCache/CoreDistributedCacheProvider.cs

```diff
@@ -0,0 +1,160 @@
+#region License
+
+//
+//  CoreDistributedCache - A cache provider for NHibernate using Microsoft.Extensions.Caching.Distributed.IDistributedCache.
+//
+//  This library is free software; you can redistribute it and/or
+//  modify it under the terms of the GNU Lesser General Public
+//  License as published by the Free Software Foundation; either
+//  version 2.1 of the License, or (at your option) any later version.
+//
+//  This library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+//  Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public
+//  License along with this library; if not, write to the Free Software
+//  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Text;
+using Microsoft.Extensions.Caching.Distributed;
+using NHibernate.Cache;
+using NHibernate.Util;
+
+namespace NHibernate.Caches.CoreDistributedCache
+{
+	/// <summary>
+	/// Cache provider using the System.Runtime.Caching classes
+	/// </summary>
+	public class CoreDistributedCacheProvider : ICacheProvider
+	{
+		private static readonly Dictionary<string, IDictionary<string, string>> ConfiguredCachesProperties;
+		private static readonly INHibernateLogger Log;
+
+		/// <summary>
+		/// The <see cref="ICoreDistributedCacheFactory"/> factory to use for getting <see cref="IDistributedCache"/>
+		/// instances. By default, its value is initialized from <c>factory-class</c> attribute of the
+		/// <c>coredistributedcache</c> configuration section.
+		/// </summary>
+		/// <remarks>
+		/// Changes to this property affect only caches built after the change.
+		/// </remarks>
+		public static ICoreDistributedCacheFactory CacheFactory { get; set; }
```

Comment:
See [this comment](https://github.com/nhibernate/NHibernate-Caches/issues/28#issuecomment-374648421) on this PR matching issue.